android/src/main/java/com/doublesymmetry/trackplayer/service/MusicService.kt

```diff
@@ -381,7 +385,7 @@ class MusicService : HeadlessJsTaskService() {
         }
 
         scope.launch {
-            event.onMediaSessionCallbackTriggered.collect {
+            event.onPlayerActionTriggeredExternally.collect {
```

Comment:
This observer should now be collecting events when media session OR notification triggers player events. Make sure to put some breakpoints here and test it.